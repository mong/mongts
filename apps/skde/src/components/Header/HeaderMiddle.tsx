import React from "react";
import { Toolbar, Typography, styled, Container } from "@mui/material";
import Grid from "@mui/material/Grid2";
import { Breakpoint } from "@mui/material";
import { useRouter } from "next/router";

const StyledToolbar = styled(Toolbar)(({ theme }) => ({
  paddingTop: theme.spacing(12),
  paddingBottom: theme.spacing(8),
}));

export type HeaderData = {
  title: string;
  subtitle: string;
};

type HeaderMiddleProps = {
  headerData: HeaderData;
  children?: React.ReactNode;
  bgcolor?: string;
  maxWidth?: false | Breakpoint;
};

export const HeaderMiddle = (props: HeaderMiddleProps) => {
  const router = useRouter();
  const mainUrl = "https://apps.skde.no" + router.asPath;

  return (
    <StyledToolbar
      sx={{
        bgcolor: process.env.NEXT_PUBLIC_VERIFY
          ? "error.main"
          : props.bgcolor || "primary.light",
      }}
      className="header-middle"
    >
      <Container maxWidth={props.maxWidth} disableGutters={true}>
        <Grid container spacing={2} rowSpacing={6}>
          <Grid size={{ xs: 12 }}>
            {process.env.NEXT_PUBLIC_VERIFY && (
              <>
                <Typography variant="h6">
                  OBS! Dette er en verifiseringsside for testdata. Hvis du
                  ønsker å se offisielle tall må du gå til{" "}
                  <a href={mainUrl} target="_self">
                    www.skde.no.
                  </a>
                </Typography>
                <br />
              </>
            )}
            <Typography variant="h1">{props.headerData.title}</Typography>
          </Grid>
          <Grid size={{ xs: 12 }}>
            <Typography variant="h6">{props.headerData.subtitle}</Typography>
          </Grid>
          {props.children !== undefined && props.children}
        </Grid>
      </Container>
    </StyledToolbar>
  );
};
