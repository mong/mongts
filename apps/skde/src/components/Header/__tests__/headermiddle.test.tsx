import React from "react";
import { render } from "@testing-library/react";
import { HeaderMiddle } from "../HeaderMiddle";

import { vi, it, describe, expect } from "vitest";

vi.mock("next/router", () => ({
  useRouter: () => ({
    asPath: "/test-path",
  }),
}));

describe("HeaderMiddle component", () => {
  it("renders with default props", () => {
    const { container } = render(<HeaderMiddle title="Default Title" />);
    expect(container).toMatchSnapshot();
  });

  it("renders with custom title and children", () => {
    const { container } = render(
      <HeaderMiddle title="Custom Title" children="Custom Children" />,
    );
    expect(container).toMatchSnapshot();
  });

  it("renders with custom background color", () => {
    const { container } = render(
      <HeaderMiddle title="Custom Title" bgcolor="custom.bgcolor" />,
    );
    expect(container).toMatchSnapshot();
  });

  it("renders with custom maximum width", () => {
    const { container } = render(
      <HeaderMiddle title="Custom Title" maxWidth="lg" />,
    );
    expect(container).toMatchSnapshot();
  });

  it("renders with NEXT_PUBLIC_VERIFY environment variable set", () => {
    process.env.NEXT_PUBLIC_VERIFY = "true";
    const { container } = render(<HeaderMiddle title="Custom Title" />);
    expect(container).toMatchSnapshot();
    process.env.NEXT_PUBLIC_VERIFY = undefined;
  });

  it("renders with NEXT_PUBLIC_VERIFY environment variable not set", () => {
    const { container } = render(<HeaderMiddle title="Custom Title" />);
    expect(container).toMatchSnapshot();
  });
});
