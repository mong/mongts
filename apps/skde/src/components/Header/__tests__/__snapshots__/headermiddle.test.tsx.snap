// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`HeaderMiddle component > renders with NEXT_PUBLIC_VERIFY environment variable not set 1`] = `
<div>
  <div
    class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular header-middle css-1hs65z2-MuiToolbar-root"
  >
    <div
      class="MuiContainer-root MuiContainer-maxWidthLg MuiContainer-disableGutters css-ho17r4-MuiContainer-root"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-1cbysey-MuiGrid-root"
      >
        <div
          class="MuiGrid-root css-vj1n65-MuiGrid-root"
        >
          <h6
            class="MuiTypography-root MuiTypography-h6 css-1miy0lu-MuiTypography-root"
          >
            OBS! Dette er en verifiseringsside for testdata. Hvis du ønsker å se offisielle tall må du gå til
             
            <a
              href="https://apps.skde.no/test-path"
              target="_self"
            >
              www.skde.no.
            </a>
          </h6>
          <br />
          <h1
            class="MuiTypography-root MuiTypography-h1 css-10l7o3s-MuiTypography-root"
          >
            Custom Title
          </h1>
        </div>
        <div
          class="MuiGrid-root css-vj1n65-MuiGrid-root"
        >
          <h6
            class="MuiTypography-root MuiTypography-h6 css-1miy0lu-MuiTypography-root"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`HeaderMiddle component > renders with NEXT_PUBLIC_VERIFY environment variable set 1`] = `
<div>
  <div
    class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular header-middle css-1hs65z2-MuiToolbar-root"
  >
    <div
      class="MuiContainer-root MuiContainer-maxWidthLg MuiContainer-disableGutters css-ho17r4-MuiContainer-root"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-1cbysey-MuiGrid-root"
      >
        <div
          class="MuiGrid-root css-vj1n65-MuiGrid-root"
        >
          <h6
            class="MuiTypography-root MuiTypography-h6 css-1miy0lu-MuiTypography-root"
          >
            OBS! Dette er en verifiseringsside for testdata. Hvis du ønsker å se offisielle tall må du gå til
             
            <a
              href="https://apps.skde.no/test-path"
              target="_self"
            >
              www.skde.no.
            </a>
          </h6>
          <br />
          <h1
            class="MuiTypography-root MuiTypography-h1 css-10l7o3s-MuiTypography-root"
          >
            Custom Title
          </h1>
        </div>
        <div
          class="MuiGrid-root css-vj1n65-MuiGrid-root"
        >
          <h6
            class="MuiTypography-root MuiTypography-h6 css-1miy0lu-MuiTypography-root"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`HeaderMiddle component > renders with custom background color 1`] = `
<div>
  <div
    class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular header-middle css-bvelj9-MuiToolbar-root"
  >
    <div
      class="MuiContainer-root MuiContainer-maxWidthLg MuiContainer-disableGutters css-ho17r4-MuiContainer-root"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-1cbysey-MuiGrid-root"
      >
        <div
          class="MuiGrid-root css-vj1n65-MuiGrid-root"
        >
          <h1
            class="MuiTypography-root MuiTypography-h1 css-10l7o3s-MuiTypography-root"
          >
            Custom Title
          </h1>
        </div>
        <div
          class="MuiGrid-root css-vj1n65-MuiGrid-root"
        >
          <h6
            class="MuiTypography-root MuiTypography-h6 css-1miy0lu-MuiTypography-root"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`HeaderMiddle component > renders with custom maximum width 1`] = `
<div>
  <div
    class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular header-middle css-1di5y52-MuiToolbar-root"
  >
    <div
      class="MuiContainer-root MuiContainer-maxWidthLg MuiContainer-disableGutters css-ho17r4-MuiContainer-root"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-1cbysey-MuiGrid-root"
      >
        <div
          class="MuiGrid-root css-vj1n65-MuiGrid-root"
        >
          <h1
            class="MuiTypography-root MuiTypography-h1 css-10l7o3s-MuiTypography-root"
          >
            Custom Title
          </h1>
        </div>
        <div
          class="MuiGrid-root css-vj1n65-MuiGrid-root"
        >
          <h6
            class="MuiTypography-root MuiTypography-h6 css-1miy0lu-MuiTypography-root"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`HeaderMiddle component > renders with custom title and children 1`] = `
<div>
  <div
    class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular header-middle css-1di5y52-MuiToolbar-root"
  >
    <div
      class="MuiContainer-root MuiContainer-maxWidthLg MuiContainer-disableGutters css-ho17r4-MuiContainer-root"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-1cbysey-MuiGrid-root"
      >
        <div
          class="MuiGrid-root css-vj1n65-MuiGrid-root"
        >
          <h1
            class="MuiTypography-root MuiTypography-h1 css-10l7o3s-MuiTypography-root"
          >
            Custom Title
          </h1>
        </div>
        <div
          class="MuiGrid-root css-vj1n65-MuiGrid-root"
        >
          <h6
            class="MuiTypography-root MuiTypography-h6 css-1miy0lu-MuiTypography-root"
          >
            Custom Children
          </h6>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`HeaderMiddle component > renders with default props 1`] = `
<div>
  <div
    class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular header-middle css-1di5y52-MuiToolbar-root"
  >
    <div
      class="MuiContainer-root MuiContainer-maxWidthLg MuiContainer-disableGutters css-ho17r4-MuiContainer-root"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-1cbysey-MuiGrid-root"
      >
        <div
          class="MuiGrid-root css-vj1n65-MuiGrid-root"
        >
          <h1
            class="MuiTypography-root MuiTypography-h1 css-10l7o3s-MuiTypography-root"
          >
            Default Title
          </h1>
        </div>
        <div
          class="MuiGrid-root css-vj1n65-MuiGrid-root"
        >
          <h6
            class="MuiTypography-root MuiTypography-h6 css-1miy0lu-MuiTypography-root"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
